cpp/basix/finite-element.cpp

```diff
@@ -74,6 +74,14 @@ constexpr int num_transformations(cell::type cell_type)
 }
 } // namespace
 //-----------------------------------------------------------------------------
+basix::FiniteElement basix::create_element(std::string family, std::string cell,
```

Comment:
Could we remove this function and just have the version that takes an enum?

Reply:
As different elements might need to be passed different enums (eg Nedelec might want a polynomial space type to take moments against (legendre, chebyshev, etc), rather than a point arrangement) so I thought it could be neater to overload `create_element` and throw errors when the wrong additional arguments are passed to an element.

We may be able to get rid of some though, as having string and element::family/cell::type versions of each feels like it might be too much

Reply:
Let's boil it down to the minimum interface - can always add interfaces later if required.

Reply:
We have this issue already though, as Lagrange spaces need a lattice::type, and other elements don't. It feels bad for `create_element` to always take this input, but for it to have no effect for other spaces

Reply:
Happy to have extra interfaces that for extra arguments - let's just remove redundant interfaces, e.g. remove strings in favour of enums.